shell/platform/android/io/flutter/embedding/android/AndroidKeyProcessor.java

```diff
@@ -5,127 +5,10 @@
 package io.flutter.embedding.android;
 
 import android.view.KeyCharacterMap;
-import android.view.KeyEvent;
-import android.view.View;
-import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
-import io.flutter.Log;
-import io.flutter.embedding.engine.systemchannels.KeyEventChannel;
-import io.flutter.plugin.editing.TextInputPlugin;
-import java.util.ArrayDeque;
-import java.util.Deque;
-import java.util.Iterator;
 
-/**
- * A class to process key events from Android, passing them to the framework as messages using
- * {@link KeyEventChannel}.
- *
- * <p>A class that sends Android key events to the framework, and re-dispatches those not handled by
- * the framework.
- *
- * <p>Flutter uses asynchronous event handling to avoid blocking the UI thread, but Android requires
- * that events are handled synchronously. So, when a key event is received by Flutter, it tells
- * Android synchronously that the key has been handled so that it won't propagate to other
- * components. Flutter then uses "delayed event synthesis", where it sends the event to the
- * framework, and if the framework responds that it has not handled the event, then this class
- * synthesizes a new event to send to Android, without handling it this time.
- */
-public class AndroidKeyProcessor {
-  private static final String TAG = "AndroidKeyProcessor";
-
-  @NonNull private final KeyEventChannel keyEventChannel;
-  @NonNull private final TextInputPlugin textInputPlugin;
+/** A class for handling combining characters in a {@link KeyEvent} stream. */
+class AndroidKeyProcessor {
```

Comment:
Might need a name change. This class now only handles dead keys, and only has an `applyCombiningCharacterToBaseCharacter` method.

Reply:
Could we eliminate this class/file, and just incorporate the functionality in one of the other classes?

Reply:
According to @dkwingsmt we're going to add a different type of primary responder so I thought we might want to keep this around in case it's useful in the other responder as well?